src/components/cesium_map/cesium_UI.js

```diff
@@ -366,6 +368,16 @@ class CesiumMap extends React.Component {
         setCamera({ facet: "Map", ...viewer.currentView.viewDict });
       };
     }, REFRESH_TIME_MS);
+
+    // store the users' viewpoint
+    this.viewpoint = setInterval(() => {
+      const loading = document.getElementById("loading").style.display;
+      if (loading && JSON.stringify(viewer.currentView.viewDict) !== JSON.stringify(preView)) {
+        console.log(JSON.stringify(viewer.currentView.viewDict) !== JSON.stringify(preView));
+        preView = viewer.currentView.viewDict;
+        setCamera({ facet: "Map", ...viewer.currentView.viewDict });
+      }
+    }, [VIEWPOINT_TIME_MS])
```

Comment:
@datadavev This might be superfluous. The above interval only updates the viewpoint when the viewpoints have a big change every 5 seconds so if the users only move the camera in a small area and do not trigger the condition, the viewpoint would not update. The new interval would check if the viewpoint has any change like `heading` or `pitch` and then update the viewpoint every 2 seconds. The new interval can guarantee the viewpoint is accurate.